vars/runPCT.groovy

```diff
@@ -0,0 +1,133 @@
+            // TODO validate plugin list exists
+            // Start validation
+            metadata = readYaml(file: metadataFile)?.pct
+            if (metadata == null) {
+                echo "Skipping PCT execution because the metadata file does not contain a pct section"
+                skipExecution = true
+                return
+            }
+            if (metadata == 'default') {
+                echo "Using default configuration for PCT"
+                metadata = [:]
+            }
+            // Allow to override pctUrl and pctRevision from metadata file
+            pctUrl = metadata.pctUrl ?: pctUrl
+            pctRevision = metadata.pctRevision ?: pctRevision
+            isPublishedImage = pctUrl.startsWith("docker://")
+            isLocalPCT = pctUrl.startsWith("file://")
+
+            // Allow override of jenkins version from metadata file
+            jenkins = metadata.jenkins ?: jenkins
+
+            if (!(isPublishedImage || isLocalPCT)) {
+                echo 'Checking connectivity to PCT sources…'
+                sh "git ls-remote --exit-code -h ${pctUrl}"
+            }
+            // Validation ended
+
+            // Jenkins war
+            infra.stashJenkinsWar(jenkins)
+            // PCT
+            // Only validate PCT if not running a remote docker image
+            if (!isPublishedImage) {
+                if (isLocalPCT) { // Deal with already existing PCT sources
+                    pctSourcesFolder = pctUrl - "file://"
+                } else {
+                    dir(pctSourcesFolder) {
+                        checkout changelog: true, poll: false, scm: [$class           : 'GitSCM', branches: [[name: pctRevision]],
+                                                                     userRemoteConfigs: [[url: pctUrl]]]
+                    }
+                }
+                dir(pctSourcesFolder) {
+                    // We may need to run things in parallel later, so avoid several checkouts
+                    stash name: "pct"
+                }
+            }
+        }
+    })
+
+    if (skipExecution) {
+        return
+    }
+
+    infra.ensureInNode(env, env.RUN_PCT_DOCKER_NODE ?: 'docker', {
+
+        stage("Running PCT") {
+            if (!isPublishedImage) {
+                dir("pctSources") {
+                    unstash "pct"
+                    pctContainerImage = docker.build("jenkins/pct:${env.BUILD_NUMBER}")
+                }
+            } else {
+                def tag = pctUrl - "docker://"
+                pctContainerImage = docker.image(tag)
+            }
+
+            def plugins = metadata.plugins
+
+            def localSnapshots = metadata.useLocalSnapshots != null ? metadata.useLocalSnapshots : true
+
+            def testingBranches = [:]
+            def warAbsolutePath = pwd() + "/jenkins.war"
+            def containerArgsBase = "-v /var/run/docker.sock:/var/run/docker.sock -v ${warAbsolutePath}:/pct/jenkins.war:ro -u root"
```

Comment:
@raul-arabaolaza Here you do not pass the Maven Settings property to `run-sh`, and PCT will be running without Azure mirror configured. IMHO we need to fix it before merge